pkg/templates/notificationtemplates/notification_generator_test.go

```diff
@@ -0,0 +1,97 @@
+package notificationtemplates_test
+
+import (
+	"testing"
+
+	"github.com/codeready-toolchain/host-operator/pkg/templates/assets"
+	"github.com/codeready-toolchain/host-operator/pkg/templates/notificationtemplates"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func TestGetNotificationTemplate(t *testing.T) {
+	logf.SetLogger(zap.Logger(true))
+	t.Run("ok", func(t *testing.T) {
+		t.Run("get notification templates", func(t *testing.T) {
+			// when
+			template, found, err := notificationtemplates.GetNotificationTemplate("userprovisioned", notificationtemplates.WithEmptyCache())
```

Comment:
instead of using the `WithEmptyCache` option, you could use a function that sets the global var to nil and call it using `defer` - see this test for an inspiration https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/cluster/cache_whitebox_test.go#L20

this will ensure that every test cleans the global variable up

Reply:
sure, i thought the options was a nicer way of doing this. since it's just an option, it doesn't need to be passed in and hence doesn't need to be used in production. it also allows me to keep the tests in a tests package, rather than in the same package as the code. however, both ways achieve the same outcome. i'll replace it with a reset func

Reply:
the templates map used to cache the results should not be public/exposed, should it? So this is why I suggested with `WithEmptyCache` option at first.

Reply:
> the templates map used to cache the results should not be public/exposed, should it? 

not at all, but that can be solved by a "whitebox test" :-)